xarray/core/dataarray.py

```diff
@@ -3819,6 +3819,209 @@ def idxmax(
             keep_attrs=keep_attrs,
         )
 
+    def argmin(
+        self,
+        dim: Union[Hashable, Sequence[Hashable]] = None,
+        axis: Union[int, None] = None,
+        keep_attrs: bool = None,
+        skipna: bool = None,
+    ) -> Union["DataArray", Dict[Hashable, "DataArray"]]:
```

Comment:
In theory, it would be nice to use either `TypeVar` or `@overload` so tools like mypy can find out the type of the the return value from `argmin()` based on the type of `dim`.

But definitely don't worry about that now, we can save that for a follow-up :)